server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -680,27 +682,43 @@ public static class CheckpointState implements Writeable {
          */
         long globalCheckpoint;
         /**
-         * whether this shard is treated as in-sync and thus contributes to the global checkpoint calculation
+         * whether this shard is treated as in-sync and has localTranslog, it contributes to the global checkpoint calculation.
```

Comment:
Based on the comment describing the field, if `inSync` is true, will `localTranslog` be true as well?

Reply:
The comment is not making it clear.
`whether this shard is treated as in-sync and has localTranslog` - means the `inSync` field denoted if a shard is in-sync and has localTranslog.

As I understand, what you mean here is: 
`whether this shard is treated as in-sync. If shard is in-sync and has local translog, it contributes to the global checkpoint calculation`

Reply:
Ack.

Reply:
Have added more in the java doc @mch2